pkg/controller/config/config_controller.go

```diff
@@ -233,6 +233,22 @@ func (r *ReconcileConfig) reconcileInstall(req reconcile.Request, res *op.Config
 			Version: tektonVersion})
 		return reconcile.Result{}, err
 	}
+
+	for _, path := range res.Spec.AddOns {
+		extension, err := mf.NewManifest(filepath.Join("deploy", "resources", path), true, r.client)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		err = extension.Transform(tfs...)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		err = extension.ApplyAll()
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	}
+
```

Comment:
shouldn't there be an `uninstallation part` when, user removes the `addons`?

Reply:
When the CR is removed, it is uninstalled automatically because of the ownership. I added code to handle adding and removing addons when  the CR is updated.